internal/compiler-interface/src/main/java/xsbti/AnalysisCallback.java

```diff
@@ -11,54 +11,171 @@
 import java.io.File;
 
 public interface AnalysisCallback {
+    /**
+     * Set the source file mapped to a concrete {@link AnalysisCallback}.
+     * @param source Source file mapped to this instance of {@link AnalysisCallback}.
+     */
     void startSource(File source);
-    /** Called to indicate that the class <code>sourceClassName</code> depends on the class
-     * <code>onClassName</code>.  Note that only classes defined in source files included in the current compilation will
-     * passed to this method.  Dependencies on classes generated by sources not in the current compilation will
-     * be passed as binary dependencies to the `binaryDependency` method.
-     * <code>context</code> gives information about the context in which this dependency has been extracted.
-     * See xsbti.DependencyContext for the list of existing dependency contexts. */
-    void classDependency(String onClassName, String sourceClassName, DependencyContext context);
-    /** Called to indicate that the class <code>fromClassName</code> depends on a
-     * class named <code>onBinaryClassName</code> from class or jar file <code>binary</code>.
-     * The <code>onBinaryClassName</code> is a binary name where inner classes are separated with $. Check Java
-     * Language Specification 13.1 for details on binary name format.
-     * <code>context</code> gives information about the context in which this dependency has been extracted.
-     * See xsbti.DependencyContext for the list of existing dependency contexts. */
-    void binaryDependency(File onBinary, String onBinaryClassName, String fromClassName, File fromSourceFile, DependencyContext context);
-    /** Called to indicate that the source file <code>source</code> produces a class file at
-     * <code>module</code> contain class <code>name</code>.*/
-    void generatedNonLocalClass(File source, File classFile, String binaryClassName, String srcClassName);
+
+    /**
+     * Indicate that the class <code>sourceClassName</code> depends on the
+     * class <code>onClassName</code>.
+     *
+     * Note that only classes defined in source files included in the current
+     * compilation will passed to this method. Dependencies on classes generated
+     * by sources not in the current compilation will be passed as binary
+     * dependencies to the `binaryDependency` method.
+     *
+     * @param onClassName Class name being depended on.
+     * @param sourceClassName Dependent class name.
+     * @param context The kind of dependency established between
+     *                <code>onClassName</code> and <code>sourceClassName</code>.
+     *
+     * @see xsbti.api.DependencyContext
+     */
+    void classDependency(String onClassName,
+                         String sourceClassName,
+                         DependencyContext context);
+
+    /**
```

Comment:
I think that `library` gives a completely different meaning. Also, a binary name doesn't come from a library necessarily, it can come from a user project that has already been compiled. I really think `binary` is the right term to use here, at least from Zinc's perspective, and we should keep it. Users should understand Zinc notation to use the API, and for that we can better document what binary means. But, since we're in the JVM, binary is the right choice because it also tells the user that we're conforming to JLS, which has the nicety that the user knows what's the supposed format of this name.